### PR TITLE
lvm: implement non-thinpool based storage pools

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
Closes #3160.

File: doc/configuration.md [hunk 1]
```diff
@@ -396,6 +396,7 @@ source                          | string    | -
 volume.block.filesystem         | string    | block based driver (lvm)          | ext4              | Filesystem to use for new volumes
 volume.block.mount\_options     | string    | block based driver (lvm)          | discard           | Mount options for block devices
 lvm.thinpool\_name              | string    | lvm driver                        | LXDPool           | Thin pool where images and containers are created.
+lvm.use\thinpool                | bool      | lvm driver                        | true              | Whether the storage pool uses a thinpool for logical volumes.
```
Comment (stgraber) [inline doc/configuration.md]: missing underscore in the key name

File: doc/storage-backends.md [hunk 2]
```diff
@@ -115,6 +115,16 @@ lxc storage create pool1 btrfs source=/dev/sdX
 
  - Uses LVs for images, then LV snapshots for containers and container snapshots.
  - The filesystem used for the LVs is ext4 (can be configured to use xfs instead).
+ - Per default, all LVM storage pools use an LVM thinpool in which logical
```
Comment (stgraber) [inline doc/storage-backends.md]: By default

File: doc/configuration.md (hunk 1 above)
Comment (brauner) [inline doc/configuration.md]: done.

File: doc/storage-backends.md (hunk 2 above)
Comment (brauner) [inline doc/storage-backends.md]: done.

**Pull request merged**